src/Http/lib.mo

```diff
@@ -25,104 +24,108 @@ module {
+          let placeholder = deps._Assets.getPlaceholder();
+          switch (placeholder.payloadUrl) {
+            case (?payloadUrl) return _redirect(payloadUrl);
+            case (null) return _collectionInfo();
           };
         };
-        case (_) {};
+
+        let metadata = deps._Tokens.getTokenData(tokenId)!;
+        let assetid : Nat = Nat32.toNat(Utils.blobToNat32(metadata));
+        let asset : AssetTypes.AssetV2 = deps._Assets.get(assetid);
+        return _processAsset(request, asset);
       };
-      switch (_getParam(request.url, "asset")) {
-        case (?atext) {
-          switch (Utils.natFromText(atext)) {
-            case (?assetid) {
-              let asset : AssetTypes.AssetV2 = deps._Assets.get(assetid);
-              // start custom
-              switch (_processAsset(request, asset)) {
-                case (?response) {
-                  return response;
-                };
-                case (null) {};
-              };
-              // end custom
-              return _processFile(Nat.toText(assetid), asset.payload, asset.payloadUrl);
-            };
-            case (_) {};
-          };
-        };
-        case (_) {};
+
+      // by asset id (same as token id??)
+      // https://.../?asset=<assetId>
+      ignore do ? {
+        let atext = _getParam(request, "asset")!;
+        let assetid = Utils.natFromText(atext)!;
+        let asset : AssetTypes.AssetV2 = deps._Assets.get(assetid);
+        return _processAsset(request, asset);
       };
 
       /**********************
       * TOKEN INDEX LOOKUP *
+      * * * * * * * * * * *
+      * https://.../<tokenIndex>
+      * https://.../<tokenIndex>?type=metadata
+      * https://.../<tokenIndex>?type=thumbnail
       **********************/
 
-      // check if there's only on "argument" to it
+      let path = Iter.toArray(Text.tokens(request.url, #text("/")));
+
       if (path.size() == 1) {
-        let parts = Iter.toArray(Text.split(path[0], #text("?")));
-        // try and convert it to a Nat from Text
-        switch (Utils.natFromText(parts[0])) {
-          // if that works, use that
-          case (?tokenIndex) {
-            switch (deps._Tokens.getTokenDataFromIndex(Nat32.fromNat(tokenIndex))) {
-              case (?assetIdBlob) {
-                let assetid : Nat = Nat32.toNat(Utils.blobToNat32(assetIdBlob));
-                let asset : AssetTypes.AssetV2 = deps._Assets.get(assetid);
-                // start custom
-                switch (_processAsset(request, asset)) {
-                  case (?response) {
-                    return response;
-                  };
-                  case (null) {};
-                };
-                // end custom
-                return _processFile(Nat.toText(assetid), asset.payload, asset.payloadUrl);
-              };
-              case (_) {};
-            };
-          };
-          case (_) {};
+        ignore do ? {
```

Comment:
@letmejustputthishere  Should we add reveal check and show placeholder here too?